src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -36,14 +37,16 @@ internal class TelemetryConfigurationOptionsSetup : IConfigureOptions<TelemetryC
             IEnumerable<ITelemetryInitializer> initializers,
             IEnumerable<ITelemetryModule> modules,
             IEnumerable<ITelemetryProcessorFactory> telemetryProcessorFactories,
-            IEnumerable<ITelemetryModuleConfigurator> telemetryModuleConfigurators)
+            IEnumerable<ITelemetryModuleConfigurator> telemetryModuleConfigurators,
+            IApplicationIdProvider applicationIdProvider)
         {
             this.applicationInsightsServiceOptions = applicationInsightsServiceOptions.Value;
             this.initializers = initializers;
             this.modules = modules;
             this.telemetryProcessorFactories = telemetryProcessorFactories;
             this.telemetryModuleConfigurators = telemetryModuleConfigurators;
             this.telemetryChannel = serviceProvider.GetService<ITelemetryChannel>();
+            this.applicationIdProvider = applicationIdProvider;
```

Comment:
lets do it like done for telemetry channel in the line above, so that if user removes the AppIdProvider from DI, it wont throw.